src/Makefile

```diff
@@ -25,6 +25,9 @@ ifeq ($(OPTIMIZATION),-O3)
 	endif
 	REDIS_LDFLAGS+=-O3 -flto
 endif
+ifneq ($(OPTIMIZATION),-O0)
+	REDIS_CFLAGS+=-fno-omit-frame-pointer
```

Comment:
This means that you're keeping frame pointers even when optimization is enabled.

This defeats the purpose of optimization.

In some environments, such as embedded systems or high-performance computing clusters, maximizing performance may be a higher priority than ease of debugging. 

The release binaries should obey the optimization principles. 

We can always provide the separate debug binaries for debugging or specific kind of testing.

you can always introduce separate build flag, not tied with optimization. Thanks.

Reply:
Moreover, this link also mentions the same - https://fedoraproject.org/wiki/Changes/fno-omit-frame-pointer

"This Change will be implemented via a %_include_frame_pointers macro to allow packages to trivially opt-out of retaining frame pointers during compilation if needed by undefining the macro. The Change owners kindly request packagers to track opt-outs in Bugzilla and blocking against our [tracking bug](https://bugzilla.redhat.com/show_bug.cgi?id=IncludeFramePointers) so that regressions can be appropriately investigated (and hopefully resolved)."

Reply:
Thank you for your message. We have discuss on these matters within #12861 . I seek to provide further elucidation:

1. The "-fomit-frame-pointer" feature is encompassed within the GCC's O1 compilation option, albeit not obligatory. My understanding posits that this feature was initially instated as default in the antiquated GCC3 over two decades ago. Given the noteworthy evolution in the landscape of contemporary software applications, adherence to this setting may not always be imperative.
Unlike the bygone era thirty years prior when computer clusters were predominantly reserved for scientific computations, contemporary cloud-native microservices heavily allocate resources to I/O operations. Redis similarly follows this trend, as evidenced by the findings in the Filipe post, illustrating that the presence or absence of the frame pointer does not impact Redis performance. I speculate that certain large-scale computational services may still benefit from this optimization option.

2. The occurrence of running Redis within embedded environments beset by register scarcity and diminished performance is exceedingly rare. Modern CPUs such as x64, ARMv9, and RISC-V flaunt an ample abundance of registers; the utilization of super pipelining technology effectively mitigates the cost of pushing instructions onto the stack to a negligible extent.

3. In production environments hosting tens of thousands or even millions of Redis instances, observability emerges as an increasingly crucial facet.

Hence, this delineates my rationale for advocating the default removal of this optimization option within Redis.

Reply:
It would still be nice to have a way to opt out, e.g. using REDIS_CFLAGS like this:

    REDIS_CFLAGS=-fomit-frame-pointer make

It wouldn't be that hard to make sure there's still a way. Historically, REDIS_CFLAGS was never touched by the Makefile itself, so the user could always override everything using this variable.